packages/data-stores/src/wpcom-plans-ui/selectors.ts

```diff
@@ -2,6 +2,17 @@ import { PlanSlug } from '@automattic/calypso-products';
 import type { State } from './reducer';
 
 export const isDomainUpsellDialogShown = ( state: State ) => !! state.showDomainUpsellDialog;
-export const getSelectedStorageOptionForPlan = ( state: State, planSlug: PlanSlug ) =>
-	state.selectedStorageOptionForPlans?.[ planSlug ];
-export const getSelectedStorageOptions = ( state: State ) => state.selectedStorageOptionForPlans;
+export const getSelectedStorageOptionForPlan = (
+	state: State,
+	planSlug: PlanSlug,
+	siteId?: number | null
+) => {
+	// @ts-expect-error TS is unhappy if we index by a null or an undefined value. We, however,
+	// expect siteId to be null or undefined here before site creation ( like during onboarding ).
+	return state.selectedStorageOptionForPlans?.[ siteId ][ planSlug ];
```

Comment:
@ddc22 To maintain the representation of a non-existent siteId as `null` or `undefined`, we now index state by `null` or `undefined` values. This gives me pause, but since we're doing this intentionally, and since JS coerces these false-y values into strings before indexing, I think it's fine?

Lemme know if you have thoughts about this.